src/MockPlayerBridge.ts

```diff
@@ -41,23 +73,157 @@ export class MockPlayerBridge extends AbstractPlayerBridge {
+  }
+
+  protected setDisplay(display: DisplayMode) {
+    return this.setDisplayState(display);
+  }
+
+  protected setMuted(muted: boolean) {
+    return this.setMutedState(muted);
+  }
+
+  protected submitUserFeedback(payload: UserFeedbackPayload) {
+    return payload;
+  }
+
+  protected subscribeConfig(listener: (value: Config) => void) {
+    const config: Config = {
+      controls: {
+        cast: true,
+        contact: true,
+        error: true,
+        fullscreen: true,
+        mute: true,
+        pip: true,
+        play: true,
+        quality: true,
+        scrubber: true,
+      },
+      customerId: 'id',
+      streamPhase: 'PRE',
+      streamPhases: ['LIVE', 'POST', 'PRE'],
+      tenantId: 'id',
+    };
+
+    setTimeout(() => {
+      const newConfig = { ...config };
+      newConfig.controls = {
+        ...newConfig.controls,
+        cast: false,
+        scrubber: false,
+      };
+      newConfig.streamPhase = 'LIVE';
+      listener(newConfig);
+    }, 1500);
+    setTimeout(() => {
+      const newConfig = { ...config };
+      newConfig.controls = {
+        ...newConfig.controls,
+        cast: true,
+        scrubber: true,
+      };
+      newConfig.streamPhase = 'POST';
+      listener(newConfig);
+    }, 3000);
+
+    listener(config);
+    return config;
+  }
+
+  protected subscribeDisplay(listener: (value: DisplayMode) => void) {
+    document.addEventListener(MockListenerEvents.DISPLAY, () => {
+      listener(this.display);
+    });
+    return this.display;
+  }
+
+  protected subscribeError(listener: (error: string | undefined) => void) {
+    listener('error');
+    return 'error';
+  }
+
+  protected subscribeMode(listener: (mode: PlaybackMode) => void) {
+    listener('LIVE');
+    return 'LIVE' as PlaybackMode;
+  }
+
+  protected subscribeMuted(listener: (value: boolean) => void) {
+    document.addEventListener(MockListenerEvents.MUTED, () => {
+      listener(this.muted);
+    });
+    return this.muted;
+  }
+
+  protected subscribePlaybackState(
+    listener: (playbackState: PlaybackState) => void,
+  ) {
+    document.addEventListener(MockListenerEvents.PLAYBACK, () => {
+      listener(this.playbackState);
+    });
+    return this.playbackState;
+  }
+
+  protected subscribeQualities(listener: (value: Qualities) => void) {
+    const buildQuality = (index: number) => ({
+      audio: {
+        bandwidth: index,
+      },
+      label: `${index}`,
+      video: {
+        bandwidth: index,
+        height: index,
+        width: index,
+      },
+    });
+    const buildQualities = (index: number) => ({
+      active: index,
+      list: [buildQuality(1), buildQuality(2), buildQuality(3)],
+      selected: index,
     });
-    return !!document.fullscreenElement;
+    const qualities = [1, 2, 3].map((index) => buildQualities(index));
+
+    setTimeout(() => listener(qualities[1]), 1500);
+    setTimeout(() => listener(qualities[2]), 3000);
+    listener(qualities[0]);
```

Comment:
Remove this listener callback; it is useless; the return value is already passed back automatically.